boltons/pathutils.py

```diff
@@ -0,0 +1,212 @@
+        newpath = pref_bar_suff.baz
+
+    Example:
+        >>> augpath('foo.bar')
+        'foo.bar'
+        >>> augpath('foo.bar', ext='.BAZ')
+        'foo.BAZ'
+        >>> augpath('foo.bar', suffix='_')
+        'foo_.bar'
+        >>> augpath('foo.bar', prefix='_')
+        '_foo.bar'
+        >>> augpath('foo.bar', base='baz')
+        'baz.bar'
+        >>> augpath('foo.tar.gz', ext='.zip', multidot=True)
+        foo.zip
+        >>> augpath('foo.tar.gz', ext='.zip', multidot=False)
+        foo.tar.zip
+        >>> augpath('foo.tar.gz', suffix='_new', multidot=True)
+        foo_new.tar.gz
+    """
+    # Breakup path
+    orig_dpath, fname = split(path)
+    if multidot:
+        # The first dot defines the extension
+        parts = fname.split('.', 1)
+        orig_base = parts[0]
+        orig_ext = '' if len(parts) == 1 else '.' + parts[1]
+    else:
+        # The last dot defines the extension
+        orig_base, orig_ext = splitext(fname)
+    # Replace parts with specified augmentations
+    if dpath is None:
+        dpath = orig_dpath
+    if ext is None:
+        ext = orig_ext
+    if base is None:
+        base = orig_base
+    # Recombine into new path
+    new_fname = ''.join((prefix, base, suffix, ext))
+    newpath = join(dpath, new_fname)
+    return newpath
+
+
+def shrinkuser(path, home='~'):
+    """
+    Inverse of ````os.path.expanduser````
+
+    Args:
+        path (PathLike): path in system file structure
+        home (str): symbol used to replace the home path. Defaults to '~', but
+            you might want to use '$HOME' or '%USERPROFILE%' instead.
+
+    Returns:
+        PathLike: path: shortened path replacing the home directory with a tilde
+
+    Example:
+        >>> path = expanduser('~')
+        >>> assert path != '~'
+        >>> assert shrinkuser(path) == '~'
+        >>> assert shrinkuser(path + '1') == path + '1'
+        >>> assert shrinkuser(path + '/1') == join('~', '1')
+        >>> assert shrinkuser(path + '/1', '$HOME') == join('$HOME', '1')
+    """
+    path = normpath(path)
+    userhome_dpath = userhome()
+    if path.startswith(userhome_dpath):
+        if len(path) == len(userhome_dpath):
+            path = home
+        elif path[len(userhome_dpath)] == os.path.sep:
+            path = home + path[len(userhome_dpath):]
+    return path
+
+
+def expandpath(path):
+    """
+    Wrapper around expanduser and expandvars.
+
+    Less aggressive than truepath. Only expands environs and tilde. Does not
+    change relative paths to absolute paths.
+
+    Args:
+        path (PathLike): string representation of a path
+
+    Returns:
+        PathLike : expanded path
+
+    Example:
+        >>> assert normpath(expandpath('~/foo')) == join(userhome(), 'foo')
+        >>> assert expandpath('foo') == 'foo'
+    """
+    path = expanduser(path)
+    path = expandvars(path)
+    return path
+
+
+def userhome(username=None):
+    """
+    Returns the user's home directory.
```

Comment:
Maybe this docstring should be expanded a bit to specify that a user's home directory can be different than `os.path.expanduser(`~user`)`, and this handles that complexity.

Reply:
I'm actually not even sure that is the case. I think the user's home directory will always be the same as `os.path.expanduser('~user')`. The only difference is that if `user` doesn't exist, userhome will raise an error, whereas expanduser will return `~user`. (Unless you saw something to indicate that's not the case, its been awhile since I wrote that function). 

Now that I'm thinking about it, I'm not sure how useful a `userhome` function is. It seems stdliby, but I've never actually used for anything except for writing `compressuser` and testing other functions in pathutils. One option is to simply remove it and user `expanduser('~')` in the one other place where it is used in this module. On the other hand, just because I don't use it doesn't mean someone else wont find it useful.

Reply:
Hmm, I lean toward `expanduser('~')`, because that's what I've done in the past and I don't have a great desire to keep up with how OSes I'm not running are changing (unless there's really no maintained builtin).

Reply:
Sounds good. I've removed `userhome` entirely.  Best to keep things simple.